pyhms/core/initializers.py

```diff
@@ -0,0 +1,127 @@
+
+    def __call__(self, pop_size: int | None, problem: Problem):
+        return [Individual(genome=genome, problem=problem) for genome in self.sample_pop(pop_size)]
+
+
+class SeededPopInitializer(PopInitializer):
+    @abstractmethod
+    def get_seed(self, problem: Problem) -> Individual:
+        pass
+
+
+class UniformGlobalInitializer(PopInitializer):
+    def prepare_sampler(self, context: dict = None) -> None:
+        self.sampler = sample_uniform(self._bounds)
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        return self.sampler(pop_size)
+
+
+class LHSGlobalInitializer(PopInitializer):
+    def prepare_sampler(self, context: dict = None) -> None:
+        self.sampler = LatinHypercube(d=len(self._bounds))
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        sample = self.sampler.random(pop_size)
+        return linear_scale(sample, self._bounds)
+
+
+class SobolGlobalInitializer(PopInitializer):
+    def prepare_sampler(self, context: dict = None) -> None:
+        self.sampler = Sobol(d=len(self._bounds), scramble=True)
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        sample = self.sampler.random(pop_size)
+        return linear_scale(sample, self._bounds)
+
+
+class GaussianInitializer(PopInitializer):
+    def prepare_sampler(self, context: dict = None) -> None:
+        try:
+            seed_genome: np.ndarray = context["seed_genome"]
+            std_dev: float = context["std_dev"]
+        except KeyError:
+            raise ValueError("GaussianInitializer requires a seed genome and a standard deviation")
+        self.sampler = sample_normal(seed_genome, std_dev, self._bounds)
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        return self.sampler(pop_size)
+
+
+class GaussianInitializerWithSeedInject(SeededPopInitializer):
+    def prepare_sampler(self, context: dict = None) -> None:
+        try:
+            seed_ind: Individual = context["seed_ind"]
+            std_dev: float = context["std_dev"]
+            preserve_fitness: bool = context.get("preserve_fitness", True)
+        except KeyError:
+            raise ValueError("GaussianInitializer requires a seed individual and a standard deviation")
+        self.sampler = sample_normal(seed_ind.genome, std_dev, self._bounds)
+        self._preserve_fitness = preserve_fitness
+        self._seed_ind = seed_ind
+
+    def get_seed(self, problem: Problem) -> Individual:
+        if self._preserve_fitness:
+            return Individual(genome=self._seed_ind.genome, problem=problem, fitness=self._seed_ind.fitness)
+        else:
+            return Individual(genome=self._seed_ind.genome, problem=problem)
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        return self.sampler(pop_size - 1)
+
+    def __call__(self, pop_size: int | None, problem: Problem):
+        return super().__call__(pop_size, problem) + [self.get_seed(problem)]
+
+
+class InjectionInitializer(SeededPopInitializer):
```

Comment:
The code is complex and the functionality is very simple. It's confusing for me.